src/transformers/file_utils.py

```diff
@@ -2284,6 +2295,19 @@ def push_to_hub(
 
         return url
 
+    def save_checkpoint(self, checkpoint_dir, epoch):
+        if not os.path.isdir(checkpoint_dir):
+            os.mkdir(checkpoint_dir)
+        # We avoid tf.train.checkpoint or saving weights in TF format, even though that includes optimizer
+        # state for us, because it requires special handling for objects like custom losses, which we use
+        # internally and which users are likely to use too
+        weights_path = os.path.join(checkpoint_dir, "weights.h5")
+        self.save_weights(weights_path)
+        extra_data = {"epoch": epoch, "optimizer_state": self.optimizer.get_weights()}
```

Comment:
Don't we need the learning rate scheduler state as well?

Reply:
Good point, I'll add that! I'm a bit worried that we have to manually list the things we're saving - `tf.train.Checkpoint` is supposed to take care of this, but the fact that it doesn't co-operate with custom losses has been a constant headache.

Reply:
Actually, wait, no need - `save_weights` also includes the iteration count!